core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/BavetConstraintStreamScoreDirectorFactoryService.java

```diff
@@ -25,9 +31,8 @@ public ScoreDirectorType getSupportedScoreDirectorType() {
     @Override
     public Supplier<AbstractScoreDirectorFactory<Solution_, Score_>> buildScoreDirectorFactory(ClassLoader classLoader,
             SolutionDescriptor<Solution_> solutionDescriptor, ScoreDirectorFactoryConfig config) {
-        ConstraintStreamImplType constraintStreamImplType_ =
-                Objects.requireNonNullElse(config.getConstraintStreamImplType(), ConstraintStreamImplType.DROOLS);
-        if (constraintStreamImplType_ != ConstraintStreamImplType.BAVET) {
+        ConstraintStreamImplType constraintStreamImplType_ = config.getConstraintStreamImplType();
+        if (constraintStreamImplType_ == DROOLS) {
```

Comment:
In the case we add more `ConstraintStreamImplType` in the future, shouldn't this be `constraintStreamImplType_!= BAVET`?

Reply:
This needs to handle the case where the impl type is null. In that case, either BAVET or DROOLS are valid options. I decided to do it like this - if we, in the distant future, decide to add any more CS implementations, the tests will catch this.